### PR TITLE
[1.13.x] fire InputEvents

### LINKED ISSUE #5481: [1.13.2] InputEvent.KeyInputEvent not getting fired

### PR DESCRIPTION
**Pull request by 0uti**
fixes #5481

### CODE REVIEW

**Comment by BlayTheNinth** [comment]
Since we don't have `Keyboard.getEventKey()` etc. anymore, the event data passed into the patched method (key, scanCode, action, modifiers) should also be added to the `KeyInputEvent` similar to how the GUI key events have them.

**Comment by 3TUSK** [comment]
While we are on this - it might be a good idea to move those events out of `net.minecraftforge.fml.common.gameevent` package and put them into `net.minecraftforge.event`.

**Comment by 0uti** [comment]
PR is updated. Please take a look.

**Comment by 0uti** [comment]
Updated. Conflicts resolved.

**Review by mezz — APPROVED** [review]

**Review by Pfuenzle — APPROVED** [review]

**Review by EwyBoy — APPROVED** [review]

File: src/main/java/net/minecraftforge/client/event/InputEvent.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.event;
+
+import net.minecraftforge.eventbus.api.Event;
+
+public class InputEvent extends Event {
```
Comment (tterrag1098) [inline src/main/java/net/minecraftforge/client/event/InputEvent.java]: Formatting is all over the place here.

**Review by tterrag1098 — CHANGES_REQUESTED** [review]

File: src/main/java/net/minecraftforge/client/event/InputEvent.java (hunk 1 above)
Comment (0uti) [reply src/main/java/net/minecraftforge/client/event/InputEvent.java]: Formatting fixed.

**Comment by Waterpicker** [comment]
So how long until this is ready. I have a mod that needs these events.

File: src/main/java/net/minecraftforge/client/event/InputEvent.java (hunk 1 above)
Comment (0uti) [reply src/main/java/net/minecraftforge/client/event/InputEvent.java]: Please see if it fits.

**Comment by alucas** [comment]
Hi,

I'm trying to update a mod that bind chat commands to keys.
It has its own GUIs for the user to add his commands and select the keys.

It's currently using that old event to intercept the keys and run the commands. So I can't finish my port without this PR for now. Or I can use the existing keybinding system but I'm not sure if this is a good choice (at first glance you can't unregister keybinding, so it's probably not the best)

**Pull request merged**